src/api-v1.js

```diff
@@ -70,100 +70,106 @@ api.declare({
-
-      let maxWaitForCachedCopy = this.maxWaitForCachedCopy;
-
-      let startTime = new Date();
-      let x = 0;
-      while (new Date() - startTime < maxWaitForCachedCopy) {
-        debug(`Check ${x++} of ${url}`);
+  // This is the ID that we need to find a backend for
+  let incomingId = `${service}_${region}`;
+  
+  // Let's pick which backend to use.  In this case, we're looking to find the
+  // only backend known that matches the potential id
+  let backends = this.cacheManagers.filter(x => x.id === incomingId);
+  if (backends.length > 1) {
+    debug('API server is misconfigured and has more than one cachemanager with id, crashing' + incomingId);
+    // Because this should never ever happen
+    process.exit(-1);
+  } else if (backends.length === 0) {
+    debug(`${incomingId} is not known`);
+    return res.status(404).json({
+      msg: `${incomingId} is not known`,
+    });
+  } else {
+    let backend = backends[0];
+    let maxWait = this.maxWaitForCachedCopy;
+    let startTime = new Date();
+    let x = 0;
 
-        let result = await backend.getUrlForRedirect(url);
+    while (new Date() - startTime < maxWait) {
+      debug(`Check ${++x} of ${url}`);
 
-        if (result.status === 'present') {
-          debug(`${logthingy} is present`);
-          res.status(302);
-          res.location(result.url);
+      let result = await backend.getUrlForRedirect(url);
 
-          // Instead of just returning result object, we want to return only
-          // known properties.  This is to avoid possible leakage
-          let datapoint = {
-            url: url,
-            service: service,
-            region: region,
-          };
+      if (result.status === 'present') {
+        debug(`${logthingy} is present`);
+        res.status(302);
+        res.location(result.url);
 
-          debug(`Found ${url}`);
-          return res.json({
-            status: result.status,
-            url: result.url,
-          });
-        } else if (result.status === 'error') {
-          debug('Redirecting uncached copy because error occured during caching');
-          debug(result.stack || 'unknown error');
-          res.status(302);
-          res.location(url);
-          return res.json({
-            url: url,
-            msg: 'Error caching file, redirecting to original',
-          });
-        }
+        // Instead of just returning result object, we want to return only
+        // known properties.  This is to avoid possible leakage
+        let datapoint = {
+          url: url,
+          service: service,
+          region: region,
+        };
 
-        // When we have Influx set up, let's submit this datapoint
-        // to a series called 'Cloud Mirror Cache Hits'
-        await delayer(1000);
+        debug(`Found ${url}`);
+        return res.json({
+          status: result.status,
+          url: result.url,
+        });
+      } else if (result.status === 'error') {
```

Comment:
but I need to avoid the case where the value of `result.status === 'pending'`.  If I did not have the else if, I would have to explicitly handle the pending case.